solr/core/src/test/org/apache/solr/cloud/TestSegmentSorting.java

```diff
@@ -71,8 +71,10 @@ public void createCollection() throws Exception {
     final String collectionName = testName.getMethodName();
     final CloudSolrClient cloudSolrClient = cluster.getSolrClient();
     
+    final String solrConfigFileName = random().nextBoolean() ? "solrconfig-sortingmergepolicyfactory.xml" : "solrconfig-indexSort.xml";
+
     final Map<String, String> collectionProperties = new HashMap<>();
-    collectionProperties.put(CoreDescriptor.CORE_CONFIG, "solrconfig-sortingmergepolicyfactory.xml");
+    collectionProperties.put(CoreDescriptor.CORE_CONFIG, solrConfigFileName);
```

Comment:
The `solrconfig-indexSort.xml` pathway here is currently failing, likely due to the `SolrIndexSearcher` TODO-annotated area of code.

Reply:
> The `solrconfig-indexSort.xml` pathway here is currently failing, likely due to the `SolrIndexSearcher` TODO-annotated area of code.

The TODO is now done and the pathway is no longer failing then.